src/Connections/GenericHook.ts

```diff
@@ -372,7 +372,9 @@ export class GenericHookConnection extends BaseConnection implements IConnection
         }
 
         const sender = this.getUserId();
-        if (sender !== this.intent.userId) {
+        const roomMembers = await this.intent.underlyingClient.getJoinedRoomMembers(this.roomId);
```

Comment:
I think it should be best to test if the sender is in the room here instead of checking equality with the intent user Id.

Reply:
Problem with this is we don't want to hit the server with /joined_members every time we handle a webhook. In terms of how to deal with this:

My feeling is we should do an `await this.intent.ensureJoined(this.roomId)` which should make sure that the hotpath is caught. We can then catch failures and retry with an `inviteUser`, `ensureJoined`.